### ISSUE TITLE
Continue VS Code Extension Fails to Recognize AWS Bedrock Profile

### ISSUE THREAD

**Issue by hassanxelamin** [continuedev/continue#2139]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://docs.continue.dev/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS: WSL2 Ubuntu 22.04.3 LTS
- Continue: Version 0.8.47
- IDE: VSCode Version 1.92.2
- Model: Bedrock: Claude 3.5 Sonnet
- config.json:
  
{
  "models": [
    {
      "title": "Bedrock: Claude 3.5 Sonnet",
      "provider": "bedrock",
      "model": "anthropic.claude-3-5-sonnet-20240620-v1:0",
      "region": "us-west-2"
    }
  ]
}
```

### Description

The Continue extension in VSCode fails to recognize the AWS Bedrock profile specified in the ~/.aws/credentials and ~/.aws/config files. Despite the correct configuration and the profile working with the AWS CLI, the extension defaults to the default profile and produces an error stating that the bedrock profile could not be found. This issue seems to be isolated to the Continue.dev extension as other tools can successfully access the AWS credentials.

My credentials look like this:

AWS Credentials (~/.aws/credentials):
```
[default]
aws_access_key_id = MY_DEFAULT_ACCESS_KEY
aws_secret_access_key = MY_DEFAULT_SECRET_KEY

[bedrock]
aws_access_key_id = MY_BEDROCK_ACCESS_KEY
aws_secret_access_key = MY_BEDROCK_SECRET_KEY
```

AWS Config (~/.aws/config):
```
[default]
region = us-west-2
output = json

[profile bedrock]
region = us-west-2
output = json
```

Continue.dev Config (~/.continue/config.json):
```
{
  "models": [
    {
      "title": "Bedrock: Claude 3.5 Sonnet",
      "provider": "bedrock",
      "model": "anthropic.claude-3-5-sonnet-20240620-v1:0",
      "region": "us-west-2"
    }
  ]
}
```

### Log output

```Shell
[Extension Host] AWS profile with name bedrock not found in ~/.aws/credentials, using default profile
y @ console.ts:137
$logExtensionHostMessage @ mainThreadConsole.ts:39
S @ rpcProtocol.ts:458
Q @ rpcProtocol.ts:443
M @ rpcProtocol.ts:373
L @ rpcProtocol.ts:299
(anonymous) @ rpcProtocol.ts:161
B @ event.ts:1230
fire @ event.ts:1261
fire @ ipc.net.ts:652
K.onmessage @ localProcessExtensionHost.ts:378
console.ts:137 [Extension Host] CredentialsProviderError: Could not resolve credentials using profile: [default] in configuration/credentials file(s).
	at resolveProfileData (c:\Users\hassa\.vscode\extensions\continue.continue-0.9.202-win32-x64\out\extension.js:462463:13)
	at c:\Users\hassa\.vscode\extensions\continue.continue-0.9.202-win32-x64\out\extension.js:462472:14
	at async Bedrock._getCredentials (c:\Users\hassa\.vscode\extensions\continue.continue-0.9.202-win32-x64\out\extension.js:468218:18)
	at async Bedrock._streamChat (c:\Users\hassa\.vscode\extensions\continue.continue-0.9.202-win32-x64\out\extension.js:468145:29)
	at async Bedrock.streamChat (c:\Users\hassa\.vscode\extensions\continue.continue-0.9.202-win32-x64\out\extension.js:445647:30)
	at async llmStreamChat (c:\Users\hassa\.vscode\extensions\continue.continue-0.9.202-win32-x64\out\extension.js:501098:22)
	at async i.value (c:\Users\hassa\.vscode\extensions\continue.continue-0.9.202-win32-x64\out\extension.js:507581:28)
log.ts:439   ERR [Extension Host] Error handling webview message: {
  "msg": {
    "messageId": "386454b4-5c59-456c-ad7d-8bd4ff461a09",
    "messageType": "llm/streamChat",
    "data": {
      "messages": [
        {
          "role": "user",
          "content": [
            {
              "type": "text",
              "text": "d"
            }
          ]
        }
      ],
      "title": "Bedrock: Claude 3.5 Sonnet",
      "completionOptions": {}
    }
  }
}

CredentialsProviderError: Could not resolve credentials using profile: [default] in configuration/credentials file(s).
console.ts:137 [Extension Host] Error handling webview message: {
  "msg": {
    "messageId": "386454b4-5c59-456c-ad7d-8bd4ff461a09",
    "messageType": "llm/streamChat",
    "data": {
      "messages": [
        {
          "role": "user",
          "content": [
            {
              "type": "text",
              "text": "d"
            }
          ]
        }
      ],
      "title": "Bedrock: Claude 3.5 Sonnet",
      "completionOptions": {}
    }
  }
}

CredentialsProviderError: Could not resolve credentials using profile: [default] in configuration/credentials file(s).
y @ console.ts:137
$logExtensionHostMessage @ mainThreadConsole.ts:39
S @ rpcProtocol.ts:458
Q @ rpcProtocol.ts:443
M @ rpcProtocol.ts:373
L @ rpcProtocol.ts:299
(anonymous) @ rpcProtocol.ts:161
B @ event.ts:1230
fire @ event.ts:1261
fire @ ipc.net.ts:652
K.onmessage @ localProcessExtensionHost.ts:378
notificationsAlerts.ts:42 Could not resolve credentials using profile: [default] in configuration/credentials file(s).
```

**Comment by sestinj** [comment]
@hassanxelamin does this still happen if you explicitly set the "profile" property to the absolute path of the credentials file?

**Comment by hassanxelamin** [comment]
@sestinj 

Hey, thanks for helping - I figured out the issue! The extension runs on my local machine (Windows), but the AWS configs/credentials were stored on the remote machine (Linux in WSL2), therefore it couldn't be accessed.

To fix it, I had to copy the files over:

First, I had to create the .aws directory on my Windows machine because it didn’t exist:

```
mkdir -p /mnt/c/Users/YourWindowsUsername/.aws
```

Then, I copied the credentials and config files from WSL2:

```
cp ~/.aws/credentials /mnt/c/Users/YourWindowsUsername/.aws/credentials
```
```
cp ~/.aws/config/mnt/c/Users/YourWindowsUsername/.aws/config
```

Once the files were there, the extension could access the credentials, and everything worked fine.
